motionplan/constraint.go

```diff
@@ -132,41 +132,79 @@ func (c *constraintHandler) CheckConstraints(cInput *ConstraintInput) (bool, flo
 // if reportDistances is false, this check will be done as fast as possible, if true maximum information will be available for debugging.
 func newSelfCollisionConstraint(
 	frame referenceframe.Frame,
-	observationInput map[string][]referenceframe.Input,
-	collisionSpecifications []*Collision,
-	reportDistances bool,
-) (Constraint, error) {
-	return newCollisionConstraint(frame, nil, observationInput, collisionSpecifications, reportDistances)
-}
-
-// newObstacleConstraint creates a constraint that will be violated if geometries constituting the given frame ever come
-// into collision with worldState geometries outside of the collisions present for the observationInput.
-// Collisions specified as collisionSpecifications will also be ignored
-// if reportDistances is false, this check will be done as fast as possible, if true maximum information will be available for debugging.
-func newObstacleConstraint(frame referenceframe.Frame,
 	fs referenceframe.FrameSystem,
-	worldState *referenceframe.WorldState,
 	observationInput map[string][]referenceframe.Input,
 	collisionSpecifications []*Collision,
 	reportDistances bool,
 ) (Constraint, error) {
-	// TODO(rb) it is bad practice to assume that the current inputs of the robot correspond to the passed in world state
-	// the state that observed the worldState should ultimately be included as part of the worldState message
-	worldState, err := worldState.ToWorldFrame(fs, observationInput)
+	// extract inputs corresponding to the frame
+	var frameInputs []referenceframe.Input
+	var err error
+	switch f := frame.(type) {
+	case *solverFrame:
```

Comment:
We can probably eliminate this type switch or push it upwards, and have this function just take a solverFrame, as there's no way to get here without one and never will be.

Reply:
This now takes a solverframe which I think cleans things up a bit